power-adapters-data/src/main/java/com/nextfaze/poweradapters/data/CursorData.java

```diff
@@ -0,0 +1,91 @@
+package com.nextfaze.poweradapters.data;
+
+import android.database.Cursor;
+import android.support.annotation.NonNull;
+
+import java.io.Closeable;
+
+/**
+ * Simple {@link Data} implementation backed by a {@link Cursor}. The cursor can be changed or replaced at any time.
+ * The cursor will never be closed.
+ *
+ * @param <T> The type of element this data contains.
+ */
+public abstract class CursorData<T> extends Data<T> implements Closeable {
+
+    @NonNull
+    private Cursor mData;
+
+    public CursorData() {
+    }
+
+    /**
+     * @return A new element without any data from the cursor.
+     */
+    public abstract T newElement();
```

Comment:
Yeah, that might be a better idea. I originally had it that way, but then I made a subclass of a subclass of CursorData (in my database, an item had all the columns of another item plus some more), so it was easier to only have to override `newElement()` returning the new type of object, and then override `populateElement()` calling super and populating the new fields.